### ISSUE TITLE
breakfix: Login hint and Additional Scopes broken

### ISSUE THREAD

**Issue by dbranco-sdc** [greenpau/go-authcrunch#23]
When using one of this features, the separator for the "Location" when redirecting remains as "?" instead "&". The configureRedirect function should be responsible to update the separator value after adding login_hint and/or additional_scopes.

Current behaviour:
```
/auth/oauth2/MyRealm?login_hint=dbranco+nothing@byom.de?redirect_url=http://localhost:8095/somewhere?login_hint=dbranco%2Bnothing%40byom.de
```

Expected:
```
/auth/oauth2/MyRealm?login_hint=dbranco+nothing@byom.de&redirect_url=http://localhost:8095/somewhere?login_hint=dbranco%2Bnothing%40byom.de
```

**Comment by nekrondev** [comment]
Yea.. run into this issue with redirecting to expired HTML pages (see https://github.com/authp/authp.github.io/issues/38):

File: **pkg/authz/handlers/redirect.go**

Just set the "&" separator after ``login_hint`` or ``additional_scopes`` had been added.

```go
        if len(rr.Redirect.LoginHint) > 0 {
                loginHint := rr.Redirect.LoginHint
                escapedLoginHint := url.QueryEscape(loginHint)
                rr.Redirect.AuthURL = fmt.Sprintf("%s%slogin_hint=%s", rr.Redirect.AuthURL, rr.Redirect.Separator, escapedLoginHint)
                rr.Redirect.Separator = "&"
        }

        if len(rr.Redirect.AdditionalScopes) > 0 {
                additionalScopes := rr.Redirect.AdditionalScopes
                escapedAdditionalScopes := url.QueryEscape(additionalScopes)
                rr.Redirect.AuthURL = fmt.Sprintf("%s%sadditional_scopes=%s", rr.Redirect.AuthURL, rr.Redirect.Separator, escapedAdditionalScopes)
                rr.Redirect.Separator = "&"
        }
```

**Comment by greenpau** [comment]
Merged.

**Comment by greenpau** [comment]
FYI, the fix is now a part of [v1.1.15](https://github.com/greenpau/caddy-security/releases/tag/v1.1.15) release.
